src/Message/Message.php

```diff
@@ -28,9 +30,35 @@ public static function ofAssistant(?string $content = null, ?array $toolCalls =
         return new AssistantMessage($content, $toolCalls);
     }
 
-    public static function ofUser(string|Text $content, Image|string ...$images): UserMessage
+    public static function ofUser(string|ContentInterface ...$content): UserMessage
     {
-        return new UserMessage($content, ...$images);
+        Assert::minCount($content, 1, 'At least a single content part must be given.');
+
+        $text = null;
+        $images = [];
+        foreach ($content as $index => $entry) {
+            if (0 === $index) {
+                $text = $entry;
+
+                if (\is_string($text)) {
+                    $text = new Text($entry);
+                }
+
+                if (!$text instanceof Text) {
+                    throw new \InvalidArgumentException('The first content piece has to be a string or Text part.');
+                }
```

Comment:
opinion or fact? the example of Oskar demonstrated that OpenAI is fine without a text first or did i get it wrong?

Reply:
Fact for the complex parsing as we discussed before. With all those castings. So in this case more of an opinion. Generally it would also work completely without text. The text is just giving instructions on how to interpret the images in the payload.

Reply:
i guess we had a misunderstanding than - sorry! now i get your previous comment ... one sec

Reply:
Now it is the simplified version. Mapping all strings to texts. Just had to put in the `UserMessage` that when only a single text is there it will utilize the simple format for serialization instead of the complex content format.